render_hawk_2/scene_class.cs

```diff
@@ -0,0 +1,275 @@
+    public System.Collections.Generic.List<material_class> material;
+
+    public scene_class ()
+    {
+        input_text = "";
+        input_text_position = 0;
+
+        output_path = "";
+        samples_processed_index = 0;
+        samples_processed = 0;
+        ray_depth = 0;
+        image_width = 0;
+        image_height = 0;
+        radiance_clamp = 0;
+        camera = new camera_class ();
+        sky_radiance = new double[3];
+        vertex = new System.Collections.Generic.List<double[]> ();
+        triangle = new System.Collections.Generic.List<triangle_class> ();
+        sphere = new System.Collections.Generic.List<sphere_class> ();
+        material = new System.Collections.Generic.List<material_class> ();
+    }
+
+    public void load (string input_path_input)
+    {
+        input_path = input_path_input;
+
+        input_text_position = 0;
+
+        vertex.Clear ();
+        triangle.Clear ();
+        sphere.Clear ();
+        material.Clear ();
+
+        input_text = System.IO.File.ReadAllText (input_path);
+
+        output_path = text_read_line ();
+        samples_processed_index = input_text_position;
+
+        while (input_text[samples_processed_index] == '\n'
+            || input_text[samples_processed_index] == '\r'
+            || input_text[samples_processed_index] == ' ')
+        {
+            samples_processed_index++;
+        }
+
+        samples_processed = (int)text_read_number ();
+        ray_depth = (int)text_read_number ();
+        image_width = (int)text_read_number ();
+        image_height = (int)text_read_number ();
+        radiance_clamp = (double)text_read_number ();
+
+        for (int i = 0; i < 3; i++)
+        {
+            camera.position[i] = (double)text_read_number ();
+        }
+        for (int i = 0; i < 3; i++)
+        {
+            camera.vector_look[i] = (double)text_read_number ();
+        }
+        for (int i = 0; i < 3; i++)
+        {
+            camera.vector_up[i] = (double)text_read_number ();
+        }
+
+        camera.view_angle_width = (double)text_read_number ();
+        camera.view_angle_height = (double)text_read_number ();
+
+        for (int i = 0; i < 3; i++)
+        {
+            sky_radiance[i] = (double)text_read_number ();
+        }
+
+        read_vertices ();
+        read_triangles ();
+        read_spheres ();
+        read_materials ();
+        
+        camera.initialize (image_width, image_height);
+    }
+
+    public void write_samples_processed ()
+    {
+        string string1 = "";
+        string string2 = "";
+        int holder = input_text_position;
+
+        string1 = input_text.Substring (0, samples_processed_index - 1);
+
+        input_text_position = samples_processed_index;
+
+        text_read_number ();
+
+        string2 = input_text.Substring (input_text_position);
+
+        input_text_position = holder;
+
+        input_text = string1 + '\r' + samples_processed.ToString () + string2;

```

Comment:
Disgusting - samples could be stored in a separate file as JSON instead of hacking together substrings of the input.